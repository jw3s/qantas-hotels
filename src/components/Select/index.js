import React from 'react';
import styled from 'styled-components';
import Typography from '../Typography';

const { Heading } = Typography;

const Wrapper = styled.div`
  display: flex;
  align-items: center;
`;

const Dropdown = styled.select`
  display: block;
  outline: none;
  color: inherit;
  border: 1px solid #ccc;
  border-radius: 6px;
  padding: 10px;
  background-color: white;
  margin-left: 7px;
`;

type Option = { label: string, value: string };
type Props = {
  label: string,
  options: Array<Option>,
};

const Select = ({ label, options, ...props }: Props) => (
  <Wrapper>
    <Heading as="h3" mb={0}>
      {label}
    </Heading>
    <Dropdown {...props}>
      {options.map((m: Option) => (
        <option key={`${m.value}`} value={m.value}>
          {m.label}
        </option>
      ))}
    </Dropdown>
  </Wrapper>
);

export default Select;
