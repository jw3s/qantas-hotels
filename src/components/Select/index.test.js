import React from 'react';
import renderer from 'react-test-renderer';

import Select from '.';

import 'jest-styled-components';

const options = [
  {
    label: 'Price low-high',
    value: 'ASC',
  },
  {
    label: 'Price high-low',
    value: 'DESC',
  },
];

describe('<Select />', () => {
  it('<Select /> should render tree', () => {
    const tree = renderer.create(<Select label="Sort by" options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
