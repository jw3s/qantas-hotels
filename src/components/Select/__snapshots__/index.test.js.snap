// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> <Select /> should render tree 1`] = `
.c1 {
  font-size: 14px;
  margin: 0;
  margin-bottom: 0px;
  color: #000;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: block;
  outline: none;
  color: inherit;
  border: 1px solid #ccc;
  border-radius: 6px;
  padding: 10px;
  background-color: white;
  margin-left: 7px;
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    Sort by
  </h3>
  <select
    className="c2"
  >
    <option
      value="ASC"
    >
      Price low-high
    </option>
    <option
      value="DESC"
    >
      Price high-low
    </option>
  </select>
</div>
`;
